javaslang/src/test/java/javaslang/collection/AbstractSeqTest.java

```diff
@@ -1800,4 +1802,122 @@ public void withDefault() {
         assertThat(withDef.apply(-1)).isEqualTo("-1");
         assertThat(withDef.apply(3)).isEqualTo("3");
     }
+
+    // -- transpose()
+
+    @Test
+    public void shouldTransposeIfEmpty() {
+        final Seq<Seq<Integer>> actual = empty();
+        assertThat(transpose(actual)).isSameAs(actual);
+    }
+
+    @Test
+    public void shouldTransposeIfSingleValued() {
+        final Seq<Seq<Integer>> actual = of(of(0));
+        final Seq<Seq<Integer>> expected = of(of(0));
+        assertThat(transpose(actual)).isEqualTo(expected);
```

Comment:
@paplorinc We could do that. However, I checked the code. It looks good to me, the original issue should be fixed. I will move the optimization you suggested to #1800.